src/main/java/teammates/storage/sqlapi/DeadlineExtensionsDb.java

```diff
@@ -0,0 +1,107 @@
+package teammates.storage.sqlapi;
+
+import java.util.List;
+
+import org.hibernate.Session;
+
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.InvalidParametersException;
+import teammates.common.util.HibernateUtil;
+import teammates.storage.sqlentity.DeadlineExtension;
+
+import jakarta.persistence.TypedQuery;
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.criteria.Root;
+
+/**
+ * Handles CRUD operations for deadline extensions.
+ *
+ * @see DeadlineExtension
+ */
+public final class DeadlineExtensionsDb extends EntitiesDb<DeadlineExtension> {
+
+    private static final DeadlineExtensionsDb instance = new DeadlineExtensionsDb();
+
+    private DeadlineExtensionsDb() {
+        // prevent initialization
+    }
+
+    public static DeadlineExtensionsDb inst() {
+        return instance;
+    }
+
+    /**
+     * Creates a deadline extension.
+     */
+    public DeadlineExtension createDeadlineExtension(DeadlineExtension de)
+            throws InvalidParametersException {
+        assert de != null;
+
+        if (!de.isValid()) {
+            throw new InvalidParametersException(de.getInvalidityInfo());
+        }
+
```

Comment:
Need to check if we deadline extension already exists (for legacy purposes, else we break the current implementation).

Check if there is another deadline extension with same userId, feedbackSessionId.